src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -398,7 +398,7 @@ public void UnregisterMessageTarget(ActivationData activation)
             ActivationCollector.TryCancelCollection(activation);
             activationsDestroyed.Increment();
 
-            scheduler.UnregisterWorkContext(new SchedulingContext(activation));
+            scheduler.UnregisterWorkContext(activation.SchedulingContext);
```

Comment:
this looks good.
Why not do the same in RegisterMessageTarget: use activation.SchedulingContext instead of var context = new SchedulingContext(activation);